### ISSUE TITLE
[droid] RequestLoader.onLoadingFinish NullPointerException

### ISSUE THREAD

**Issue by jagregory** [calatrava/calatrava#52]
**tl;dr** In Android the `RequestLoader` sometimes fails in `onLoadingFinish` with a `NullPointerException` because the receivers haven't been registered in time for `onLoadingStart` to be called. This happens when an Ajax request is triggered immediately in a controller.
## Scenario

I have a controller which immediately fires off an ajax request on init (is that an ok thing to do? it works everywhere but Android).

``` coffeescript
example.list.controller = ({views, changePage, ajax}) ->
  ajax
    url: 'http://example.com'
    method: 'GET'
    success: (d) -> 'succeeded'
    failure: (status, message) -> 'failed'
```

This creates an `AjaxRequest` in the Android bridge, which executes roughly like so:

``` java
protected Void doInBackground(String... params) {
  try {
    context.sendBroadcast(new Intent("com.calatrava.ajax.start"));

    // execute request
  } finally {
    context.sendBroadcast(new Intent("com.calatrava.ajax.finish"));
  }

  return null;
}
```

Those `sendBroadcast` calls notify the `RequestLoader` to display a progress indicator, then hide it when the request is done.
## Issue

It seems the first `sendBroadcast(new Intent("com.calatrava.ajax.start"))` call is happening _before_ the receivers have been registered (in `RegisteredActivity.onResume`), but by the time the last `sendBroadcast(new Intent("com.calatrava.ajax.finish"))` executes the receivers have been registered.

The end result is that the `RequestLoader` never receives the start notification, and then does receive the finish and crashes because the dialog doesn't exist.
## Solution?

The easy fix is to just wrap the `dialog.hide()` call in a null check, but that's not very satisfying.

I'll try to do some more digging, but I'm wondering why the `Activity` isn't fully initialized by the time the controller javascript is executed. I suspect either multi-threading, or something is blocking when it shouldn't be.

Thoughts?
